src/main/java/org/jenkinsci/deprecatedusage/DeprecatedUsage.java

```diff
@@ -50,14 +52,22 @@ public void analyze(File pluginFile) throws IOException {
         analyzeWithClassVisitor(pluginFile, classVisitor);
     }
 
+    
     public void analyzeWithClassVisitor(File pluginFile, ClassVisitor aClassVisitor)
             throws IOException {
         // recent plugins package their classes as a jar file with the same name as the war file in
         // WEB-INF/lib/ while older plugins were packaging their classes in WEB-INF/classes/
-        try (WarReader warReader = new WarReader(pluginFile, true)) {
+        try (WarReader warReader = new WarReader(pluginFile, !includePluginLibraries)) {
             String fileName = warReader.nextClass();
             while (fileName != null) {
-                analyze(warReader.getInputStream(), aClassVisitor);
+                try {
+                    @SuppressWarnings("resource") // handled by warReader.nextClass()
+                    InputStream is = warReader.getInputStream();
+                    analyze(is, aClassVisitor);
+                } catch (Exception e) {
```

Comment:
there where a few issues along the way (requiring the bump to ASM9), before this would throw an unhelful message (what could not be scanned was unclear) and it would mean the plugin would only be partially analyzed.  now we just fail to parse the single class, but keep going trying to parse other classes.